modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaKeyIdResolverModelParser.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.loader.parser.java;
+
+import static org.mule.metadata.api.utils.MetadataTypeUtils.isEnum;
+import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
+import static org.mule.runtime.core.api.util.ClassUtils.getClassName;
+
+import org.mule.metadata.api.annotation.EnumAnnotation;
+import org.mule.metadata.api.model.BooleanType;
+import org.mule.metadata.api.model.MetadataType;
+import org.mule.runtime.api.exception.MuleRuntimeException;
+import org.mule.runtime.core.api.util.ClassUtils;
+import org.mule.runtime.extension.api.metadata.NullMetadataResolver;
+import org.mule.runtime.module.extension.internal.loader.parser.KeyIdResolverModelParser;
+import org.mule.runtime.module.extension.internal.metadata.BooleanKeyResolver;
+import org.mule.runtime.module.extension.internal.metadata.EnumKeyResolver;
+import org.mule.runtime.module.extension.internal.metadata.SdkTypeKeysResolverAdapter;
+import org.mule.sdk.api.metadata.resolving.PartialTypeKeysResolver;
+
+/**
+ * {@link KeyIdResolverModelParser} for Java based syntax
+ *
+ * @since 4.5.0
+ */
+public class JavaKeyIdResolverModelParser implements KeyIdResolverModelParser {
+
+  private final MetadataType metadataType;
+  private final String categoryName;
+  private final Class<?> keyIdResolverDeclarationClass;
+  private final boolean muleResolver;
+
+  public JavaKeyIdResolverModelParser(String categoryName, MetadataType metadataType, Class<?> keyIdResolverDeclarationClass,
+                                      boolean muleResolver) {
+    this.metadataType = metadataType;
+    this.categoryName = categoryName;
+    this.keyIdResolverDeclarationClass = keyIdResolverDeclarationClass;
+    this.muleResolver = muleResolver;
+  }
+
+  @Override
+  public boolean hasKeyIdResolver() {
+    return (!org.mule.sdk.api.metadata.NullMetadataResolver.class.isAssignableFrom(keyIdResolverDeclarationClass) &&
```

Comment:
There should be a utils to check if something is a null resolver, I think we have this logic in other places also.